backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Docs_Shipping_Notification_Description.sql

```diff
@@ -0,0 +1,89 @@
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+DROP FUNCTION IF EXISTS de_metas_endcustomer_fresh_reports.Docs_Shipping_Notification_Description(record_id   numeric,
+                                                                                                  ad_language character varying)
+;
+
+CREATE FUNCTION de_metas_endcustomer_fresh_reports.Docs_Shipping_Notification_Description(record_id   numeric,
+                                                                                          ad_language character varying)
+    RETURNS TABLE
+            (
+                DeliveryFrom          character varying,
+                DeliveryTo            character varying,
+                Document_name         character varying,
+                documentno            character varying,
+                sales_order           character varying,
+                poreference           character varying,
+                physicalclearancedate timestamp WITHOUT TIME ZONE,
+                Harvesting_year       character varying,
+                description           character varying
+            )
+    STABLE
+    LANGUAGE sql
+AS
+$$
+SELECT TRIM(COALESCE(wbp.name || E'\n', '') ||
+            COALESCE(wbploc.address1 || E'\n', '') ||
+            COALESCE(wbploc.postal || E'\n', '') ||
+            COALESCE(wbploc.city || E'\n', '') ||
+            COALESCE(wbpc.name, '')
+           )           AS DeliveryFrom,
+       TRIM(COALESCE(snbp.name || E'\n', '') ||
+            COALESCE(snbploc.address1 || E'\n', '') ||
+            COALESCE(snbploc.postal || E'\n', '') ||
+            COALESCE(snbploc.city || E'\n', '') ||
+            COALESCE(snbpc.name, '')
+           )           AS DeliveryTo,
+       dt_trl.name     AS Document_name,
+       sn.documentno,
+       o.documentno    AS sales_order,
+       sn.poreference,
+       sn.physicalclearancedate,
+       year.fiscalyear AS Harvesting_year,
+       sn.description
+
+FROM M_Shipping_Notification sn
+         -- DeliveryFrom (warehouse partner and location)
+         INNER JOIN m_warehouse w ON sn.m_warehouse_id = w.m_warehouse_id
+         INNER JOIN c_bpartner wbp ON w.c_bpartner_id = wbp.c_bpartner_id
+         INNER JOIN c_bpartner_location wbpl ON wbp.c_bpartner_id = wbpl.c_bpartner_id
+         INNER JOIN c_location wbploc ON wbploc.c_location_id = wbpl.c_location_id
+         INNER JOIN C_Country wbpc ON wbploc.C_Country_ID = wbpc.C_Country_ID
```

Comment:
in all other docs, this is comming from sub report for address.
Pls use "de/metas/docs/generics/report_bp_org_block_left.jasper"
And create a subreport for warehouse address

Reply:
added new sub report `report_delivery_from.jrxml`